### DISCUSSION TITLE
Are there standards you'd like to see included in the ecosystem?

### DISCUSSION THREAD

**Discussion by XAMPPRocky** [librasn/rasn#333] [Q&A]
Are there standards that you'd like to see but aren't currently available? Please post suggestions of standards that could become crates below 🙂

**Comment by Nicceboy** [comment]
I have been personally working on with IEEE Std 1609.2-2022, ETSI TS 103 097, and ETSI TS 102 941 quite some time, I was planning to make PR probably quite soon, just that you know it. IEEE Std 1609.2-2022 is pretty big and I have also tested correctness of the COER codec against that.

**Reply by XAMPPRocky** [reply]
That is exciting 🙂

**Reply by Nicceboy** [reply]
What could be actually the best approach now that compiler can do a lot already? Use the compiler-generated as base and fill the missing parts and utilities manually?

**Reply by XAMPPRocky** [reply]
That would be my recommendation for starting since it massively cuts down on time spent manually creating data structures.

**Comment by Nicceboy** [comment]
Some standards have massive amount of fields in structs, and most of them can be optional. Builder pattern would be useful for these.

Implementing those in hand take enormous amount of time, so that would require using procedural macros.

I have been playing around with [bon](https://github.com/elastio/bon), maybe we could use that in standards in the meantime if we ever implement on our own macros?

Builder pattern is also useful on guaranteeing the backwards compatibility for any code that uses those standards.

**Reply by XAMPPRocky** [reply]
Yeah using Bon is totally fine, as long as it's no_std dependencies are fine.
